File: core/src/main/java/com/kin/ecosystem/core/data/settings/SettingsDataSource.java
package com.kin.ecosystem.core.data.settings;

public interface SettingsDataSource {

	void setIsBackedUp(final String publicAddress, boolean isBackedUp);

	boolean isBackedUp(final String publicAddress);

	interface Local {

		void setIsBackedUp(final String publicAddress, boolean isBackedUp);

		boolean isBackedUp(final String publicAddress);
	}
}

File: core/src/main/java/com/kin/ecosystem/core/data/settings/SettingsDataSource.kt
package com.kin.ecosystem.core.data.settings

interface SettingsDataSource {

    fun setIsBackedUp(publicAddress: String, isBackedUp: Boolean)

    fun isBackedUp(publicAddress: String): Boolean

    fun hasSeenTransfer(kinUserId: String): Boolean

    fun setSeenTransfer(kinUserId: String)

    fun setSawOnboarding(kinUserId: String)

    fun hasSeenOnboarding(kinUserId: String) : Boolean

    interface Local {

        fun setIsBackedUp(publicAddress: String, isBackedUp: Boolean)

        fun isBackedUp(publicAddress: String): Boolean

        fun setSeenOnboarding(kinUserId: String)

        fun hasSeenOnboarding(kinUserId: String) : Boolean

        fun setSeenTransfer(kinUserId: String)

        fun hasSeenTransfer(kinUserId: String) : Boolean
    }
}
